diff -ur a/src/mountall.c b/src/mountall.c
--- a/src/mountall.c	2013-01-14 15:13:47.000000000 -0600
+++ b/src/mountall.c	2013-01-23 22:09:48.454768263 -0600
@@ -2700,17 +2700,8 @@
 
 	nih_assert (name != NULL);
 
-	if (mnt && cb) {
-		reply_data = NIH_MUST (nih_alloc (NULL, sizeof (EventReplyData)));
-		reply_data->mnt = mnt;
-		reply_data->handler = cb;
-	}
-
-	if (no_events) {
-		if (cb)
-			cb (reply_data, NULL);
+	if (no_events)
 		return;
-	}
 
 	/* Flush the Plymouth connection to ensure all updates are sent,
 	 * since the event may kill plymouth.
@@ -2744,6 +2735,9 @@
 		nih_discard (var);
 	}
 
+	if (mnt && cb)
+		reply_data = NIH_MUST (nih_alloc (NULL, sizeof (EventReplyData)));
+	
 	pending_call = NIH_SHOULD (upstart_emit_event (upstart,
 						       name, env, reply_data ? TRUE : FALSE,
 						       reply_data ? cb : NULL,
@@ -2754,6 +2748,9 @@
 	
 		if (reply_data) {
 
+			reply_data->mnt = mnt;
+			reply_data->handler = cb;
+
 			/* If previous event is still pending, wait for it. */
 			if (mnt->pending_call) {
 				DBusPendingCall *pending = mnt->pending_call;
