From: mk01 <matuskral@me.com>
Date: Fri, 1 Feb 2013 02:06:57 +0100
Subject: no filesystem should depend on mounting swap first

---
 src/mountall.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/mountall.c b/src/mountall.c
index e2396f6..a23f407 100644
--- a/src/mountall.c
+++ b/src/mountall.c
@@ -1345,7 +1345,8 @@ mount_policy (void)
 		    && strcmp (mnt->device, "none")
 		    && strcmp (mnt->device, mnt->type)
 		    && (mnt->entry.prev != mounts)
-		    && strcmp (((Mount *)mnt->entry.prev)->device, "none")) {
+		    && strcmp (((Mount *)mnt->entry.prev)->device, "none")
+		    && strcmp (((Mount *)mnt->entry.prev)->type, "swap")) {
 			NihListEntry *dep;
 			Mount *       prior;
 
-- 
